In [3]:
import pandas as pd
import numpy as np

In [4]:
data = pd.read_csv('소상공인시장진흥공단_전국 온누리상품권 가맹점 현황_20221013.csv', encoding='cp949')
data

가맹점명 소속 시장명(또는 상점가)                            소재지
0           철원상회         영동전통시장         서울특별시 강남구 논현동  140-22 
1          신앙촌상회         영동전통시장          서울특별시 강남구 논현동  145-1 
2            밥과술         영동전통시장         서울특별시 강남구 논현동  163-24 
3           강남축산         영동전통시장       서울특별시 강남구 논현동  145-11 1층
4           잡곡상회         영동전통시장    서울특별시 강남구 논현동  144-10  101호
...          ...            ...                            ...
101468     고운우리옷       제주로얄쇼핑센터  제주특별자치도 제주시 연동 272-33  2층 76호
101469       쉐르담       제주로얄쇼핑센터  제주특별자치도 제주시 연동 272-33 1층 113호
101470       팅커벨       제주로얄쇼핑센터    제주특별자치도 제주시 연동 272-33 1층 77
101471  행복이 있는 곳       제주로얄쇼핑센터  제주특별자치도 제주시 연동 272-33 1층 1~2호
101472    동남고추상회         표선오일시장   제주특별자치도 서귀포시 표선면 표선리 1001-1 

[101473 rows x 3 columns]

In [6]:
data['소속 시장명(또는 상점가)'].value_counts()

남대문시장               757
부산진시장               732
서문시장2지구 종합상가        706
고투몰(강남터미널지하도상점가)    581
동산상가                568
                   ... 
지보면상점가                1
부곡시장                  1
동진시장                  1
장승포신부시장               1
표선오일시장                1
Name: 소속 시장명(또는 상점가), Length: 1814, dtype: int64

In [10]:
data['가맹점점포수'] = data.groupby('소속 시장명(또는 상점가)')['가맹점명'].transform('count')
data

가맹점명 소속 시장명(또는 상점가)                            소재지  가맹점점포수
0           철원상회         영동전통시장         서울특별시 강남구 논현동  140-22      186
1          신앙촌상회         영동전통시장          서울특별시 강남구 논현동  145-1      186
2            밥과술         영동전통시장         서울특별시 강남구 논현동  163-24      186
3           강남축산         영동전통시장       서울특별시 강남구 논현동  145-11 1층     186
4           잡곡상회         영동전통시장    서울특별시 강남구 논현동  144-10  101호     186
...          ...            ...                            ...     ...
101468     고운우리옷       제주로얄쇼핑센터  제주특별자치도 제주시 연동 272-33  2층 76호      51
101469       쉐르담       제주로얄쇼핑센터  제주특별자치도 제주시 연동 272-33 1층 113호      51
101470       팅커벨       제주로얄쇼핑센터    제주특별자치도 제주시 연동 272-33 1층 77      51
101471  행복이 있는 곳       제주로얄쇼핑센터  제주특별자치도 제주시 연동 272-33 1층 1~2호      51
101472    동남고추상회         표선오일시장   제주특별자치도 서귀포시 표선면 표선리 1001-1        1

[101473 rows x 4 columns]

In [21]:
df = data.groupby('소속 시장명(또는 상점가)', as_index=False)[['가맹점명']].count()
df.rename(columns={'소속 시장명(또는 상점가)' : '시장명',
                  '가맹점명' : '가맹점수'}, inplace=True)
df

시장명  가맹점수
0           (새)인현지하상가    12
1         (유)다사소상공인연합    29
2     (주)대현프리몰 지하도상점가   120
3           (주)여수수산시장    99
4           1913송정역시장    53
...               ...   ...
1809             흑석시장     1
1810             흥국상가    78
1811          흥덕전통시장      3
1812             흥부시장     8
1813             흥해시장    63

[1814 rows x 2 columns]

In [22]:
df2 = pd.read_csv('전국전통시장표준데이터.csv', encoding='cp949')
df2 = df2[['시장명', '시장유형', '위도', '경도', '점포수', '사용가능상품권', '공중화장실보유여부', '주차장보유여부']]
df2

시장명 시장유형         위도          경도  점포수            사용가능상품권 공중화장실보유여부  \
0     동원전통종합시장  상설장  37.589824  127.092107  127                NaN         Y   
1       면목골목시장  상설장  37.577485  127.084436   93                NaN         Y   
2        사가정시장  상설장  37.581081  127.089379  135                NaN         N   
3         신아타운  상설장  37.606453  127.094707  150                NaN         Y   
4       우림골목시장  상설장  37.596772  127.098272  127                NaN         Y   
...        ...  ...        ...         ...  ...                ...       ...   
1516     ㈜중앙시장  상설장  34.740596  127.732581  131  여수지역사랑상품권, 온누리상품권         Y   
1517      ㈜서시장  상설장  34.741792  127.728651  191  여수지역사랑상품권, 온누리상품권         Y   
1518   서시장주변시장  상설장  34.740832  127.728382  233  여수지역사랑상품권, 온누리상품권         Y   
1519      성주시장  상설장  35.915025  128.285479   45             온누리상품권         Y   
1520      벽진시장  5일장  35.948766  128.216546    4                NaN         Y   

     주차장보유여부  
0          Y  
1          Y  
2          N  
3          Y  
4          Y  
...      ...  
1516       Y  
1517       Y  
1518       Y  
1519       Y  
1520       N  

[1521 rows x 8 columns]

In [23]:
merge_data = pd.merge(df, df2, how='inner', on='시장명')
merge_data

시장명  가맹점수     시장유형         위도          경도  점포수            사용가능상품권  \
0     (주)여수수산시장    99      상설장  34.738142  127.731764  115                NaN   
1       Team204    21      상설장  37.567713  127.013197  100                NaN   
2         ㈜신정시장   154      상설장  35.542336  129.310284   94   온누리상품권+울산페이+제로페이   
3         ㈜야음시장    13      상설장  35.524059  129.331372   85   온누리상품권+울산페이+제로페이   
4    ㈜여수수산물특화시장    16      상설장  34.736825  127.732856  209  여수지역사랑상품권, 온누리상품권   
..          ...   ...      ...        ...         ...  ...                ...   
991        회기시장    17      상설장  37.590970  127.054979   64                NaN   
992      회성종합시장    12      상설장  35.232929  128.559410   78                NaN   
993        효자시장   141      상설장  36.008551  129.332678  208                NaN   
994        흥국상가    78      상설장  34.762008  127.669333  268  여수지역사랑상품권, 온누리상품권   
995        흥해시장    63  상설장+5일장  36.110619  129.348288  153                NaN   

    공중화장실보유여부 주차장보유여부  
0           Y       N  
1           Y       Y  
2           Y       N  
3           Y       Y  
4           Y       Y  
..        ...     ...  
991         Y       Y  
992         Y       Y  
993         Y       Y  
994         N       N  
995         Y       Y  

[996 rows x 9 columns]

In [24]:
merge_data['가맹률'] = merge_data['가맹점수'] / merge_data['점포수'] * 100
merge_data

시장명  가맹점수     시장유형         위도          경도  점포수            사용가능상품권  \
0     (주)여수수산시장    99      상설장  34.738142  127.731764  115                NaN   
1       Team204    21      상설장  37.567713  127.013197  100                NaN   
2         ㈜신정시장   154      상설장  35.542336  129.310284   94   온누리상품권+울산페이+제로페이   
3         ㈜야음시장    13      상설장  35.524059  129.331372   85   온누리상품권+울산페이+제로페이   
4    ㈜여수수산물특화시장    16      상설장  34.736825  127.732856  209  여수지역사랑상품권, 온누리상품권   
..          ...   ...      ...        ...         ...  ...                ...   
991        회기시장    17      상설장  37.590970  127.054979   64                NaN   
992      회성종합시장    12      상설장  35.232929  128.559410   78                NaN   
993        효자시장   141      상설장  36.008551  129.332678  208                NaN   
994        흥국상가    78      상설장  34.762008  127.669333  268  여수지역사랑상품권, 온누리상품권   
995        흥해시장    63  상설장+5일장  36.110619  129.348288  153                NaN   

    공중화장실보유여부 주차장보유여부         가맹률  
0           Y       N   86.086957  
1           Y       Y   21.000000  
2           Y       N  163.829787  
3           Y       Y   15.294118  
4           Y       Y    7.655502  
..        ...     ...         ...  
991         Y       Y   26.562500  
992         Y       Y   15.384615  
993         Y       Y   67.788462  
994         N       N   29.104478  
995         Y       Y   41.176471  

[996 rows x 10 columns]

**시장별 온누리상품권 가맹점수(온누리상품권 가맹점 정보) / 시장별 점포수(전국전통시장표준 데이터**

-> 전국전통시장표준 데이터에 나와있는 시장 수(1521개)와 온누리상품권 가맹점 데이터의 시장 수(1814) 정보가 많이 겹치지 않아서 가맹률을 구할 수 있는 시장 수가 996개로 줄어듦

-> 점포수보다 가맹점수가 더 많은 경우가 있어 신뢰도 급 하락